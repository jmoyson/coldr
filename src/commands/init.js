import { createCampaign } from '../services/campaign.service.js';
import { logInfo, logSuccess, createSpinner } from '../utils/error.utils.js';
import chalk from 'chalk';

/**
 * Initialize a new campaign
 * @param {string} campaignName - Campaign name
 * @returns {string} Campaign directory path
 */
export default function init(campaignName = 'coldr-campaign') {
  const spinner = createSpinner('Creating campaign').start();
  const campaignPath = createCampaign(campaignName);
  spinner.succeed('Campaign created');

  console.log('');
  logSuccess('Campaign ready.');
  logInfo(`Folder: ${chalk.cyan(campaignPath)}`);
  console.log('');
  logInfo('Run:');
  console.log(`  ${chalk.white('export RESEND_API_KEY=re_*****')}`);
  console.log(`  ${chalk.white('coldr schedule --dry-run')}`);
  console.log('');

  return campaignPath;
}
