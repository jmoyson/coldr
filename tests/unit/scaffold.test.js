import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import fs from 'fs';
import path from 'path';
import init from '../../src/commands/init.js';

describe('Unit: Scaffold Output', () => {
  const testCampaignName = 'test-scaffold-campaign';
  const testCampaignPath = path.join(process.cwd(), testCampaignName);

  beforeEach(() => {
    // Mock console to avoid cluttering test output
    vi.spyOn(console, 'log').mockImplementation(() => {});
    vi.spyOn(console, 'error').mockImplementation(() => {});

    // Mock process.stderr.write for ora spinners
    vi.spyOn(process.stderr, 'write').mockImplementation(() => {});
  });

  afterEach(() => {
    // Cleanup: remove test campaign if it exists
    if (fs.existsSync(testCampaignPath)) {
      fs.rmSync(testCampaignPath, { recursive: true, force: true });
    }
    vi.restoreAllMocks();
  });

  it('should match the scaffold snapshot', () => {
    init(testCampaignName);

    const config = fs.readFileSync(path.join(testCampaignPath, 'config.json'), 'utf-8');
    const leads = fs.readFileSync(path.join(testCampaignPath, 'leads.csv'), 'utf-8');
    const template = fs.readFileSync(path.join(testCampaignPath, 'template.html'), 'utf-8');
    const suppressions = fs.readFileSync(path.join(testCampaignPath, 'suppressions.json'), 'utf-8');

    expect(config).toMatchSnapshot();
    expect(leads).toMatchSnapshot();
    expect(template).toMatchSnapshot();
    expect(suppressions).toMatchSnapshot();
  });
});
