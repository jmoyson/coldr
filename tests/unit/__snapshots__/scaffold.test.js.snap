// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unit: Scaffold Output > should match the scaffold snapshot 1`] = `
"{
  "sender": "Your Name <hello@sending.yourdomain.com>",
  "replyTo": "hello@yourdomain.com",
  "subject": "Quick question about {{company}}",
  "perDay": 20,
  "startDate": "2025-10-29T00:00:00Z",
  "workDays": [1, 2, 3, 4, 5],
  "workHours": [9, 17]
}
"
`;

exports[`Unit: Scaffold Output > should match the scaffold snapshot 2`] = `
"email,name,company
alice@example.com,Alice,Example Corp
bob@sample.io,Bob,Sample Inc
charlie@demo.co,Charlie,Demo Co
"
`;

exports[`Unit: Scaffold Output > should match the scaffold snapshot 3`] = `
"<p>Hi {{name}},</p>
<p>I noticed {{company}} is working on [relevant topic].</p>
<p>Quick question: would you be open to a 15-minute intro call?</p>
<p>Best,<br />Alice</p>
"
`;

exports[`Unit: Scaffold Output > should match the scaffold snapshot 4`] = `
"{
  "emails": [],
  "domains": []
}
"
`;
